vtex-sites/base.store
# feat: Add VTEX Search tracking

## What's the purpose of this pull request?
This PR integrates with VTEX Search tracking API. Also, this PR integrates two events:
1. search.query
2. search.click

These events are necessary for populating `Top searches` and adding relevance to search. 

Also, this PR adds the origin url to all analytics events so we can know which url generated each event.

## How does it work?
Search does not have a proper script, like VTEX RC or Google Analytics, but a tracking API. With that, I have implemented a client for this tracking api on the file `src/sdk/analytics/platform/vtex/search.ts`. I have roughly followed [the docs](https://www.notion.so/vtexhandbook/Event-API-Documentation-48eee26730cf4d7f80f8fd7262231f84) on which fields we send to the tracking api. 

After doing the client, I had to raise the right events and capture them to send to the tracking API. Turns out, we already had both events, 'search' and 'select_item' being raised.

`search` event was being raised inside the `doSearch` function. We don't have the necessary info inside this function, so I've moved this event to the `ProductGallery`. This change, made `search` event to be raised on the following situations:
1. When the user search a term via the searchbar.
2. When (inside the search result page), the user changes the sorting method.
3. When (inside the search result page), the user changes the filter applied.

I'm not sure if we should raise the event on cases 2,3. Make comments on the review!

Finally, I thought it would be interesting to add a url for each raised event, so our analytics engine can enrich event data with the current url the user is viewing. I see this being useful to track the urls where the user adds/remove a product to/from cart etc.

Also, make sure to review:
https://github.com/vtex/faststore/pull/1174

## How to test it?
To track search events, open the network tab of chrome, filter by `sp.vtex.com`. Them search by terms and click on products. You should see requests being made.
Make sure these events are only raised on full text pages, not on department pages or shelves.

## References
- Search tracking API Docs: https://www.notion.so/vtexhandbook/Event-API-Documentation-48eee26730cf4d7f80f8fd7262231f84

## Checklist
